source/fab/steps/compile_fortran.py

```diff
@@ -346,6 +346,42 @@ def compile_file(self, analysed_file, flags, output_fpath):
             value={'time_taken': timer.taken, 'start': timer.start})
 
 
+def get_fortran_preprocessor():
+    fpp, fpp_flags = None, None
+
+    try:
+        fpp, fpp_flags = get_tool(os.getenv('FPP'))
+        logger.info(f'env defined fpp as "{fpp}"')
+    except ValueError:
+        pass
+
+    if not fpp:
+        try:
+            run_command(['which', 'fpp'])
+            fpp, fpp_flags = 'fpp', ['-P']
```

Comment:
A note about the purpose of "-P" and a link to https://github.com/stfc/fparser/issues/390

Reply:
Is GitHub playing silly buggers? you've acknowledged this comment but I don't see a comment in the diff.

Reply:
I think you'll find it is _I_ who is the silly bugger. Done now.